.github/workflows/build-ironic-images.yaml

```diff
@@ -15,35 +15,61 @@ on:
     types: [checks_requested]
 
 jobs:
-  build-ironic-images:
+  build:
     runs-on: ubuntu-latest
-    defaults:
-      run:
-        working-directory: ironic-images/ipa-debian-bookworm
+    strategy:
+      matrix:
+        image:
+          - name: "ubuntu"
+            script: "keekz-ubuntu-extras"
+          - name: "debian"
+            script: "ipa-debian-bookworm"
     steps:
-      - uses: actions/checkout@eef61447b9ff4aafe5dcd4e0bbf5d482be7e7871 # v4
-      - uses: actions/setup-python@f677139bbe7f9c59b41e40162b753c062f5d49a3 # v5
+      - name: Checkout repository
+        uses: actions/checkout@eef61447b9ff4aafe5dcd4e0bbf5d482be7e7871 # v4
+
+      - name: Setup python environment
+        uses: actions/setup-python@f677139bbe7f9c59b41e40162b753c062f5d49a3 # v5
         id: setup-python
         with:
           python-version: '3.11'
           cache: 'pip'
-      - run: sudo apt update && sudo apt install -y debootstrap qemu-utils
-      - run: pip install -r requirements.txt
+
+      - name: Install Dependencies
+        run: sudo apt update && sudo apt install -y debootstrap qemu-utils squashfs-tools kpartx
+
+      - name: Install python packages
+        run: pip install -r requirements.txt
         working-directory: ironic-images
-      - name: Build the IPA image
-        run: bash ipa-debian-bookworm.sh
+
+      - name: Build Images
+        run: |
+          bash "${{ matrix.image.script }}".sh
+          if [ "${{ matrix.image.name }}" == "ubuntu" ]; then
+            echo "qcow2_file=ironic-images/${{ matrix.image.script }}/${{ matrix.image.script }}.qcow2" >> $GITHUB_ENV
+          elif [ "${{ matrix.image.name }}" == "debian" ]; then
+            echo "initramfs_file=ironic-images/${{ matrix.image.script }}/${{ matrix.image.script }}.initramfs" >> $GITHUB_ENV
+            echo "kernel_file=ironic-images/${{ matrix.image.script }}/${{ matrix.image.script }}.kernel" >> $GITHUB_ENV
+          fi
+        working-directory: "ironic-images/${{ matrix.image.script }}"
         env:
           ELEMENTS_PATH: ${{ env.pythonLocation }}/share/ironic-python-agent-builder/dib:${{ github.workspace }}/ironic-images/custom_elements
-      - name: Dynamically set timestamp environment variable
-        run: echo "TIMESTAMP=$(date +"%Y%m%d%H%M%S")" >> $GITHUB_ENV
-      - name: Publish IPA Release
+
+      - name: Get commit hash
+        run: echo "COMMIT_HASH=$(git rev-parse --short HEAD)" >> $GITHUB_ENV
+
+      - name: Set timestamp environment variable
+        run: echo "TIMESTAMP=$(git show --no-patch --no-notes --pretty='%cd' --date=format:'%Y%m%d%H%M%S' ${{ env.COMMIT_HASH }})" >>  $GITHUB_ENV
+
+      - name: Publish Release
```

Comment:
Now that we're doing a matrix I think we'll have to split up the release into its own step. Because this will cause two release tags to be published. So I think each build will need to use the https://github.com/actions/upload-artifact to upload the built bits. Then the create release step will need to download those artifacts and make the release from that so that there's one release with all the files.

Reply:
I used timestamp of the commit sha so that it would be one release. The previous version generated a timestamp when the job run so with two jobs, you get two timestamp and 2 releases. The image that I added to the PR shows the one release and the three built files.

Reply:
So the GitHub Actions matrix runner will run each one separately to completion. You are creating the release in the matrix so it'll make 2 releases on each run instead of one combined release.

Reply:
https://github.com/argoproj/argo-workflows/blob/main/.github/workflows/release.yaml that's an example of a matrix then coming back together to one release.